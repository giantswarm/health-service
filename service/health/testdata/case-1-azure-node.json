{
  "metadata": {
    "name": "fg0ge-master-000000",
    "selfLink": "/api/v1/nodes/fg0ge-master-000000",
    "uid": "6c0ddfe9-e6b4-11e9-8147-000d3a267334",
    "resourceVersion": "3057",
    "creationTimestamp": "2019-10-04T14:36:59Z",
    "labels": {
      "azure-operator.giantswarm.io/version": "2.6.1",
      "beta.kubernetes.io/arch": "amd64",
      "beta.kubernetes.io/instance-type": "Standard_D2s_v3",
      "beta.kubernetes.io/os": "linux",
      "failure-domain.beta.kubernetes.io/region": "westeurope",
      "failure-domain.beta.kubernetes.io/zone": "0",
      "giantswarm.io/provider": "azure",
      "ip": "10.2.0.5",
      "kubernetes.io/arch": "amd64",
      "kubernetes.io/hostname": "fg0ge-master-000000",
      "kubernetes.io/os": "linux",
      "kubernetes.io/role": "master",
      "node-role.kubernetes.io/master": "",
      "node.kubernetes.io/master": "",
      "role": "master"
    },
    "annotations": {
      "node.alpha.kubernetes.io/ttl": "0",
      "volumes.kubernetes.io/controller-managed-attach-detach": "true"
    }
  },
  "spec": {
    "podCIDR": "10.2.128.0/24",
    "providerID": "azure:///subscriptions/1be3b2e6-497b-45b9-915f-eb35cae23c6a/resourceGroups/fg0ge/providers/Microsoft.Compute/virtualMachineScaleSets/fg0ge-master/virtualMachines/0",
    "taints": [
      {
        "key": "node-role.kubernetes.io/master",
        "effect": "NoSchedule"
      }
    ]
  },
  "status": {
    "capacity": {
      "attachable-volumes-azure-disk": "4",
      "cpu": "2",
      "ephemeral-storage": "28454196Ki",
      "hugepages-1Gi": "0",
      "hugepages-2Mi": "0",
      "memory": "8145236Ki",
      "pods": "110"
    },
    "allocatable": {
      "attachable-volumes-azure-disk": "4",
      "cpu": "1400m",
      "ephemeral-storage": "27405620Ki",
      "hugepages-1Gi": "0",
      "hugepages-2Mi": "0",
      "memory": "6236500Ki",
      "pods": "110"
    },
    "conditions": [
      {
        "type": "NetworkUnavailable",
        "status": "False",
        "lastHeartbeatTime": "2019-10-04T14:38:55Z",
        "lastTransitionTime": "2019-10-04T14:38:55Z",
        "reason": "RouteCreated",
        "message": "RouteController created a route"
      },
      {
        "type": "MemoryPressure",
        "status": "False",
        "lastHeartbeatTime": "2019-10-04T14:51:36Z",
        "lastTransitionTime": "2019-10-04T14:36:48Z",
        "reason": "KubeletHasSufficientMemory",
        "message": "kubelet has sufficient memory available"
      },
      {
        "type": "DiskPressure",
        "status": "False",
        "lastHeartbeatTime": "2019-10-04T14:51:36Z",
        "lastTransitionTime": "2019-10-04T14:36:48Z",
        "reason": "KubeletHasNoDiskPressure",
        "message": "kubelet has no disk pressure"
      },
      {
        "type": "PIDPressure",
        "status": "False",
        "lastHeartbeatTime": "2019-10-04T14:51:36Z",
        "lastTransitionTime": "2019-10-04T14:36:48Z",
        "reason": "KubeletHasSufficientPID",
        "message": "kubelet has sufficient PID available"
      },
      {
        "type": "Ready",
        "status": "True",
        "lastHeartbeatTime": "2019-10-04T14:51:36Z",
        "lastTransitionTime": "2019-10-04T14:39:03Z",
        "reason": "KubeletReady",
        "message": "kubelet is posting ready status"
      }
    ],
    "addresses": [
      {
        "type": "InternalIP",
        "address": "10.2.0.5"
      },
      {
        "type": "Hostname",
        "address": "fg0ge-master-000000"
      }
    ],
    "daemonEndpoints": {
      "kubeletEndpoint": {
        "Port": 10250
      }
    },
    "nodeInfo": {
      "machineID": "89797ad1eaa34c0baa6b3f9f1bba1c1b",
      "systemUUID": "2757b7b8-0b8e-0143-ae90-b44d42b4fb32",
      "bootID": "50d7a69b-6759-4756-967b-bb9b834e1b52",
      "kernelVersion": "4.19.50-coreos-r1",
      "osImage": "Container Linux by CoreOS 2135.4.0 (Rhyolite)",
      "containerRuntimeVersion": "docker://18.6.3",
      "kubeletVersion": "v1.14.6",
      "kubeProxyVersion": "v1.14.6",
      "operatingSystem": "linux",
      "architecture": "amd64"
    },
    "volumesAttached": [
      {
          "name": "kubernetes.io/azure-disk//subscriptions/foo/resourceGroups/GS_aksprod-1_aksprod-1_westeurope/providers/Microsoft.Compute/disks/kubernetes-dynamic-pvc-sample-hash-1234567890abc",
          "devicePath": "/dev/xvdbk"
      }
    ]
  }
}