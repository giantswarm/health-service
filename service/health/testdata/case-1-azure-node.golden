{
    "identity": {
        "name": "fg0ge-master-000000",
        "kubelet_version": "v1.14.6",
        "operator_version": "",
        "role": "master",
        "ip": "10.2.0.5",
        "hostname": "fg0ge-master-000000",
        "instance_type": "Standard_D2s_v3",
        "availability_region": "westeurope",
        "availability_zone": "0"
    },
    "machine": {
        "cpu_count": 2,
        "memory_capacity_bytes": 8340721664,
        "memory_allocatable_bytes": 6386176000,
        "ephemeral_storage_capacity_bytes": 29137096704,
        "ephemeral_storage_allocatable_bytes": 28063354880,
        "attachable_volume_allocatable_count": 4,
        "attachable_volume_capacity_count": 4,
        "attached_volumes_count": 1
    },
    "ready": true,
    "health": "green"
}